chaco/grid.py

```diff
@@ -333,31 +333,59 @@ def _draw_overlay(self, gc, view_bounds=None, mode="normal"):
 
         Overrides PlotComponent.
         """
-        self._draw_component(gc, view_bounds, mode)
+        # What we're really trying to do with a grid is plot contour lines in
+        # the space of the plot.  In a rectangular plot, these will always be
+        # straight lines.
+        if not self.visible:
+            return
 
-    def overlay(self, other_component, gc, view_bounds=None, mode="normal"):
-        """Draws this component overlaid on another component.
+        if not self._cache_valid:
+            self._compute_ticks()
 
-        Overrides AbstractOverlay.
-        """
-        if not self.visible:
+        if len(self._tick_positions) == 0:
             return
-        self._compute_ticks(other_component)
-        self._draw_component(gc, view_bounds, mode)
-        self._cache_valid = False
 
-    def _draw_component(self, gc, view_bounds=None, mode="normal"):
```

Comment:
same comment as before regarding the duplication

Reply:
like before, you can make `overlay` call `_draw_overlay`

Reply:
Here they are _very slightly_ different... 😕 
eg `overlay` calls `self._compute_ticks(other_component)` but `_draw_overlay` does not...